File: javascript/madlib/madlib.js
import { input } from "tinyinput"

async function main(){
    const noun = await input("Enter a noun: ", "string")
    const verb = await input("Enter a verb: ", "string")
    const adjective = await input("Enter an adjective: ", "string")

    console.log(`The ${noun} was given a choice to do ${verb} in a way that is ${adjective}`)
}

main()

File: go/madlib/madlib.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strings"
)

func main(){
	fmt.Print("Enter a noun,verb and adjective: ")
	reader := bufio.NewReader(os.Stdin)
	words, err := reader.ReadString('\n')
	if err != nil {
		log.Fatal(err)
	}

	words = strings.TrimSpace(words)
	word_list := strings.Split(words, ",")

	if len(word_list) != 3 {
		log.Fatal("words should be 3 in number")
	}

	fmt.Printf("The %s was doing %s and that person was doing it %s\n", word_list[0], word_list[1], word_list[2])
}